compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -141,7 +141,7 @@ class TreeChecker extends Phase with SymTransformer {
       override def apply(parent: Tree, tree: Tree)(using Context): Tree = {
         tree match {
           case tree: New if !parent.isInstanceOf[tpd.Select] =>
-            assert(assertion = false, i"`New` node must be wrapped in a `Select`:\n  parent = ${parent.show}\n  child = ${tree.show}")
+            assert(assertion = false, i"`New` node must be wrapped in a `Select` of the constructor:\n  parent = ${parent.show}\n  child = ${tree.show}")
```

Comment:
Maybe the message could mention how to do this for the reflection API? Is there an equivalent to `nme.CONSTRUCTOR`? Or a convenience method? Also maybe we should also check that if a Select exists, then the name selected has `isConstructorName` return true.